src/openrct2/actions/RideCreateAction.cpp

```diff
@@ -154,10 +154,25 @@ GameActions::Result RideCreateAction::Execute() const
     ride->vehicle_change_timeout = 0;
     ride->num_stations = 0;
     ride->num_vehicles = 1;
-    ride->proposed_num_vehicles = 32;
+    if (gCheatsDisableTrainLengthLimit)
+    {
+        // Reduce amount of proposed trains to prevent 32 trains from always spawning when limits are disabled
+        if (rideEntry->cars_per_flat_ride == 0xFF)
```

Comment:
Could you perhaps introduce a constant for the 0xFF value, considering it has a special meaning, and replace the existing occurrences?